core/src/main/java/com/opensymphony/xwork2/util/fs/DefaultFileManager.java

```diff
@@ -40,14 +40,25 @@
     private static final Pattern JAR_PATTERN = Pattern.compile("^(jar:|wsjar:|zip:|vfsfile:|code-source:)?(file:)?(.*?)(\\!/|\\.jar/)(.*)");
     private static final int JAR_FILE_PATH = 3;
 
-    protected static Map<String, Revision> files = Collections.synchronizedMap(new HashMap<String, Revision>());
+    protected static final Map<String, Revision> files = Collections.synchronizedMap(new HashMap<String, Revision>());
+    private static final List<URL> lazyMonitoredFilesCache = Collections.synchronizedList(new ArrayList<URL>());
```

Comment:
`equals` implementation of `URL` will resolved hosts to compare if they have the same IP, when comparing two URLs two requests are made to the external world. Maybe this is not the case here (we use files) but I'm trying to avoid using `URL`s directly.